Source/Orts.Simulation/Simulation/RollingStocks/MSTSSteamLocomotive.cs

```diff
@@ -2349,6 +2386,49 @@ private void UpdateTender(float elapsedClockSeconds)
             {
                 WaterIsExhausted = false;
             }
+
+            // update water scoop
+            // Water scoop fill charateristics can be found in - 
+            // Calculate water velocity entering pipe: v = SQRT ( loco speed^2 - 2 * gravity * h)
+            // Calculate the drag of the water scoop in the water: Drag Force = 0.5 * Drag Coeff * Fluid Density * Reference Area * Velocity
+            
+            if (HasWaterScoop && IsWaterScoopDown && RefillingFromTrough && Direction == Direction.Forward && AbsSpeedMpS > WaterScoopMinSpeedMpS)
+            {
+            // Calculate water velocity
+               float Avalue = ((float)Math.Pow(MpS.ToMpH(absSpeedMpS), 2) * 2.15f);
+               float Bvalue = 2.0f * 32.2f * Me.ToFt(WaterScoopFillElevationM);
+               
+               if (Avalue > Bvalue)
+               {
+                  WaterScoopVelocityMpS = Me.FromFt((float)Math.Sqrt(Avalue - Bvalue));
+               }
+               else
+               {
+                  WaterScoopVelocityMpS = 0;
+               }
+               
+               // calculate volume of water scooped per period
+               float CuFttoGalUK = 6.22884f; // imperial gallons of water in a cubic foot of water
```

Comment:
We should be using metric units everywhere internally to avoid confusion.

Reply:
I agree ideally.

However given the age of steam locomotives, they are typically engineered in imperial units. Hence a lot of the relevant formula are in imperial units, and I have not found many metric formulas. I have chosen to keep formulas in their original forms so that they can be easily verified against source reference material. Also saves me from introducing errors by trying to convert an imperial formula to its metric equivalent.

Wherever possible I have taken the outputs of these formula and converted them to metric units for usage elsewhere in OR.

Reply:
As long as we're keeping the use of non-metric units and conversions to a minimum, that's okay. Every conversion introduces a small error to the calculations, so we need to avoid repeatedly converting the same value, as well.